domain/agentbinary/types.go

```diff
@@ -0,0 +1,18 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package agentbinary
+
+import (
+	"github.com/juju/juju/core/objectstore"
+)
+
+// Metadata holds the metadata of the agent binary to be stored into the database.
+type Metadata struct {
```

Comment:
I think introducing this type might be a bit pre-mature just yet as we haven't full got the design for the parts that will be using this.

Reply:
As discussed, this is just a placeholder for the empty state method.
It can be removed or changed during state implementation.